From 4908eb5d27fce4cb5ffdb0f72b08352c4e4fe637 Mon Sep 17 00:00:00 2001
From: t <tristan75015@gmail.com>
Date: Mon, 12 Oct 2026 01:12:24 +0200
Subject: [PATCH] ex2, ex6: behaviour library and device options

- ex2: option -l (--library), so that the behaviour may be given by a
  library generated by the generic-parallel interface, as in ex4.
- ex6: options of mfem-mgis (declareDefaultOptions), in particular the
  device (--device); description of -l as in ex4.
---
 ex2/ssna303.cxx             | 3 +++
 ex6/RveNonLinearElastic.cxx | 6 +++++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/ex2/ssna303.cxx b/ex2/ssna303.cxx
index 70aeb68..9a3927b 100644
--- a/ex2/ssna303.cxx
+++ b/ex2/ssna303.cxx
@@ -133,6 +133,9 @@ int main(int argc, char** argv) {
                  "boundary, no comparison if empty.");
   args.AddOption(&parallel, "-p", "--parallel", "-no-p", "--no-parallel",
                  "Perform parallel computations.");
+  args.AddOption(&library, "-l", "--library",
+                 "Behaviour library (for instance a library generated by "
+                 "the generic-parallel interface).");
 #ifdef MGIS_HAVE_TFEL
   args.AddOption(&use_fbar, "-fb", "--use-fbar", "-no-fb", "--no-use-fbar",
                  "Use the FBar formulation.");
diff --git a/ex6/RveNonLinearElastic.cxx b/ex6/RveNonLinearElastic.cxx
index ddc852c..26a21cb 100644
--- a/ex6/RveNonLinearElastic.cxx
+++ b/ex6/RveNonLinearElastic.cxx
@@ -47,7 +47,9 @@ struct TestParameters {
 
 void common_parameters(mfem::OptionsParser& args, TestParameters& p) {
   args.AddOption(&p.mesh_file, "-m", "--mesh", "Mesh file to use.");
-  args.AddOption(&p.library, "-l", "--library", "Material library.");
+  args.AddOption(&p.library, "-l", "--library",
+                 "Behaviour library (for instance a library generated by "
+                 "the generic-parallel interface).");
   args.AddOption(&p.order, "-o", "--order",
                  "Finite element order (polynomial degree).");
   args.AddOption(&p.refinement, "-r", "--refinement",
@@ -59,6 +61,8 @@ void common_parameters(mfem::OptionsParser& args, TestParameters& p) {
                  "two-layer cube, only valid for cube_2mat_per.mesh.");
   args.AddOption(&p.verbosity_level, "-v", "--verbosity-level",
                  "Verbosity level of the solvers.");
+  // options of mfem-mgis, such as the device (--device)
+  mfem_mgis::declareDefaultOptions(args);
 
   args.Parse();
 
-- 
2.43.0

